File: data structure and algo/Have some code data structures/searching and sorting/2k19 attempt/quicksort.cpp
#include<iostream>
using namespace std;

int partition(int arr[], int l, int h){
	int pivot,index;
	index = l;
	pivot = h;
	for (int i = l; i<h; i++){
		if(arr[i]<a[pivot]){
			swap(&arr[i], &arr[index]);
			index++
		}
	}
	swap(&arr[pivot], &arr[index]);
	return index;
}

int quicksort(int arr[], int l, int h){
	int pivot = partition(arr,l,h);
	quicksort(arr,l,pivot-1);
	quicksort(arr, pivot, h);
}
int main(){
	
}

File: data structure and algo/Have some code data structures/searching and sorting/2k20 attempts/quick sort.py
import random
arr = [int(i) for i in input().split(' ')]

def helper(arr, s, e):
    i = s-1
    j = s
    n = e-1
    pivot = arr[e]
    # i is denoting the end index region which is smaller than pivot
    for j in range(s, e-1):
        if(arr[j] < pivot):
            i += 1
            arr[i], arr[j] = arr[j], arr[i]
        else:
            j += 1
    arr[j], arr[i+1] = arr[i+1], arr[j]
    return i+1

def shuffling(arr):
    random.shuffle(arr)

def quickSort(arr, s, e):
    pivotIndex = helper(arr, s, e)
    quickSort(arr, s, pivotIndex-1)
    quickSort(arr, pivotIndex+1, e)


shuffling(arr)
print(arr)